##' Processing SRA file to RNA-Seq read count table
##' 
##' 
##' SRA files, downloaded automatically from NCBI, are first converted into "FASTQ" file 
##' using SRA Toolkit (version 2.3.5-2). Alignments were done by \code{align()} and read counts 
##' were generated by \code{featureCounts()}. Both functions are in R package \code{Rsubread}
##'  (version 1.14.2). The reference genome is "Arabidopsis_thaliana.TAIR10.22.gtf", available at
##'  Ensembl plants FTP server (http://plants.ensembl.org/ info/data/ftp/index.html)
##'
##'
##' @note require loading 
##'
##' @title pro
##'
##' @param y an n-vector of NB counts.
##' @param s an n-vector of library sizes (multiplicative offset).
##' @param x an n by p design matrix.
##' @param phi a scalar or an n-vector, the NB dipsersion parameter.
##' @param beta0 a p-vector specifying the known and unknown
##' components of beta, the regression coefficients. NA values
##' indicate unknown components and non-NA values specify the values
##' of the known components. The default is that all components of
##' beta are unknown.
##' @param ... furhter arguements to be passed to \code{\link{irls.nb.1}}.
##' @return a list
##'   \item{mu}{an n-vector, estimated means (MLE).}
##'   \item{beta}{an p-vector, estimated regression coefficients (MLE).}
##'   \item{iter}{number of iterations performed in the IRLS algorithm.}
##'   \item{zero}{logical, whether any of the estimated \code{mu} is close to zero.}
##'   \item{l}{log likelihood of the fitted model.}
##'   \item{D}{a p-vector, the score vector}
##'   \item{i}{a p-by-p matrix, fisher information matrix}




library(Rsubread)
library(edgeR)
library(limma)

datafile <- "GSE32216"

setwd("/Users/Bin/Dropbox/Zhuo/Research/Project2014/Code/RunInfo")
read.file <- paste(datafile, "SraRunInfo.csv", sep="" )
RunInfo <- read.csv(read.file, stringsAsFactors=F)
id <- which(RunInfo$LibraryStrategy=="RNA-Seq" & RunInfo$LibrarySelection=="cDNA"
            &RunInfo$LibraryLayout=="SINGLE")
RunInfo <- RunInfo[id, ]



(fs <- basename(RunInfo$download_path))
filepath <- RunInfo$download_path
fsname <- RunInfo$SampleName
fastq.name <- RunInfo$Run
layout <- RunInfo$LibraryLayout


for (i in 1:length(fs))
{
  download.file(url= filepath[i], destfile = paste("/home/zhuob/protocol/Rsubread/", fs[i], sep=""))
}


for (i in 1:length(fs))
{
  
  path="/home/zhuob/protocol/software/sratoolkit.2.3.5-2-ubuntu64/bin/"
  fastq <- paste(path, "fastq-dump --split-3", sep="")
  
  fileSRA <- paste("/home/zhuob/protocol/Rsubread/", fs[i], sep="")
  cmd <- paste(fastq, fileSRA)
  cat(cmd, "\n")
  system(cmd)
}


 buildindex(base="arab_index", reference="Arabidopsis_thaliana.TAIR10.22.dna.toplevel.fa")

for (i in fastq.name)
{
  file1 <- paste(i, ".fastq", sep="")
  out <- paste(i, ".bam", sep="")
  
  align(index="arab_index", readfile1=file1, input_format ="FASTQ",  output_file=out)
  remove.file <- paste("/home/zhuob/protocol/Rsubread/", file1, sep="")
  cmd <- paste("rm", file1)
  system(cmd)
}



bamfile <- paste(fastq.name, ".bam", sep="")

## The options for featureCounts are discussed here. see #78
# http://seqanswers.com/forums/showthread.php?t=30258&page=4-

fc <- featureCounts(files=bamfile, annot.ext="Arabidopsis_thaliana.TAIR10.22.gtf",
                    isGTFAnnotationFile=T)

cun <- fc$counts
cun <- data.frame(cun[order(row.names(cun)),])
colnames(cun) <- fsname

dest.path <- "/home/zhuob/Dropbox/Zhuo/Research/Project2014/Data/arab/Rsubread/"

dest.file <- paste(dest.path, datafile, ".Rsubread.txt", sep="")
write.table(cun, dest.file)





